django/middleware/http.py

```diff
@@ -19,6 +19,10 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        # It's too late to prevent an unsafe request with a 412 response.
+        if request.method not in ('GET', 'HEAD'):
```

Comment:
Your comment actually made me realize that CGM can't work on a `HEAD` request either. Updating the ticket and PR...